client/src/components/FilesDialog/FilesDialog.vue

```diff
@@ -1,16 +1,334 @@
+<script setup lang="ts">
+import { library } from "@fortawesome/fontawesome-svg-core";
+import { faCaretLeft } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/vue-fontawesome";
+import { BAlert, BButton } from "bootstrap-vue";
+import Vue, { computed, onMounted, ref } from "vue";
+
+import { UrlTracker } from "@/components/DataDialog/utilities";
+import { isSubPath } from "@/components/FilesDialog/utilities";
+import { selectionStates } from "@/components/SelectionDialog/selectionStates";
+import { errorMessageAsString } from "@/utils/simple-error";
+
+import { getGalaxyInstance } from "../../app";
```

Comment:
```suggestion
import { getGalaxyInstance } from "@/app";
import { UrlTracker } from "@/components/DataDialog/utilities";
import { isSubPath } from "@/components/FilesDialog/utilities";
import { selectionStates } from "@/components/SelectionDialog/selectionStates";
import { errorMessageAsString } from "@/utils/simple-error";

```

Reply:
Thanks for bringing this up! I forgot to remove the `getGalaxyInstance` entirely and replace it with the configuration composable. I will do that instead 👍

Reply:
Argh... the store injection doesn't seem to work on this component so `useConfig` does not work... I tried to "provide" the store in https://github.com/galaxyproject/galaxy/blob/dev/client/src/utils/data.js#L82 but I couldn't get it to work... Anyone knows a workaround for this? or should I just revert to using `getGalaxyInstance` for this...?

Reply:
@davelopez I can't see the problem. Can you explain it?

I tested it out and the component gets the config object correctly.

Reply:
You need to have an FTP set up in your Galaxy, try to access it (it would do weird things) and look at the console:


https://github.com/galaxyproject/galaxy/assets/46503462/16f1519e-16eb-44b4-a983-5c637f15298f

Reply:
I decided to go all in and directly convert the config store to Pinia in https://github.com/galaxyproject/galaxy/pull/16514 to avoid these issues (and probably transform them into new ones... :rofl:)